frontend/public/components/secrets/create-secret.tsx

```diff
@@ -0,0 +1,155 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+import { k8sCreate, k8sUpdate, K8sResourceKind } from '../../module/k8s';
+import { ButtonBar, Firehose, history, kindObj, StatusBox } from '../utils';
+import { getActiveNamespace, formatNamespacedRouteForResource, UIActions } from '../../ui/ui-actions';
+import { SafetyFirst } from '../safety-first';
+import { WebHookSecretSubform } from './subforms';
+
+const determineSecretTypeAbstraction = (secret) => {
+  const { data } = secret;
+  if (_.has(data, 'WebHookSecretKey')) {
+    return 'webhook';
+  }
+  return 'generic';
+};
+
+class BaseEditSecret_ extends SafetyFirst<BaseEditSecretProps_, BaseEditSecretState_> {
+  constructor (props) {
+    super(props);
+    let existingData = _.pick(props.obj, ['metadata', 'data']);
+    const secret = _.defaultsDeep({}, props.fixed, existingData, {
+      apiVersion: 'v1',
+      data: {},
+      kind: 'Secret',
+      metadata: {
+        name: '',
+      },
+      type: 'Opaque',
+    });
+
+    this.state = {
+      secretType: this.props.secretType || determineSecretTypeAbstraction(secret),
+      secret: secret,
+      inProgress: false,
+      type: secret.type,
+      stringData: _.mapValues(secret.data, (v) => window.atob(v)),
+    };
+    this.onNameChanged = this.onNameChanged.bind(this);
+    this.save = this.save.bind(this);
+  }
+  onDataChanged (secretsData) {
+    let stringData = {...this.state.stringData};
+    stringData = secretsData;
+    this.setState({stringData});
+  }
+  onNameChanged (event) {
+    let secret = {...this.state.secret};
+    secret.metadata.name = event.target.value;
+    this.setState({secret});
+  }
+  save (e) {
+    e.preventDefault();
+    const { kind, metadata } = this.state.secret;
+    this.setState({ inProgress: true });
+    const newSecret = _.assign({}, this.state.secret, {stringData: this.state.stringData});
```

Comment:
Does this include both `data` and `stringData`? We should only include `stringData`